app/wheatley.py

```diff
@@ -0,0 +1,287 @@
+        if 'stop_at_rounds' not in self._settings:
+            self._settings['stop_at_rounds'] = False
+
+        # If there is no row_gen defined, set it to even-bell Plain Bob on the right number of bells
+        if self.row_gen == {}:
+            self._set_default_row_gen()
+
+        self._is_active = False
+        self._wheatley_process = None
+        self._thread = None
+        self._should_thread_stop = False
+
+        self._has_been_closed = False
+
+    def log(self, message):
+        """ Function to make a log entry under this Wheatley instance's name """
+        log(f"WHEATLEY:{self._tower.tower_id}:{message}")
+
+    @property
+    def enabled(self):
+        """ Returns True if Wheatley is enabled """
+        return self._enabled
+
+    def set_enabledness(self, value):
+        """
+        Set the enabledness of Wheatley, making sure that the Tower's user list is kept up-to-date.
+        """
+        self._enabled = value
+        # Add or remove 'Wheatley' to the tower's user list
+        if self._enabled:
+            self._tower.add_user(USER_ID, "Wheatley")
+        else:
+            self._tower.remove_user(USER_ID)
+
+    # ===== SETTINGS =====
+
+    @property
+    def settings(self):
+        """ Return Wheatley's settings. """
+        return self._settings
+
+    def update_settings(self, new_settings):
+        """ Merge some changed settings into Wheatley's settings. """
+        # Copy only the fields from new_settings into the dict
+        for key, value in new_settings.items():
+            self._settings[key] = value
+
+    # ===== ROW GEN =====
+
+    @property
+    def row_gen(self):
+        """ Returns the JSON representation of the row generator that Wheatley will use. """
+        return self._row_gen
+
+    @row_gen.setter
+    def row_gen(self, value):
+        """ Sets the JSON representation of the row generator that Wheatley will use. """
+        self._row_gen = value
+
+    def _set_default_row_gen(self):
+        """ Sets the row generator to some value that is guarunteed to be valid. """
+        stage = self._tower.n_bells
+        stage_name = _get_stage_name(stage)
+        plain_bob_notation = {
+            4: 'x14x14,12',
+            6: 'x16x16x16,12',
+            8: 'x18x18x18x18,12',
+            10: 'x10x10x10x10x10,12',
+            12: 'x1Tx1Tx1Tx1Tx1Tx1T,12'
+        }[stage]
+
+        self._row_gen = {
+            'type': 'method',
+            'title': 'Plain Bob ' + stage_name,
+            'url': 'Plain_Bob_' + stage_name,
+            'stage': stage,
+            'notation': plain_bob_notation,
+            'bob': {'0': '14'},
+            'single': {'0': '1234'}
+        }
+
+    # ===== CALLBACKS =====
+
+    def on_assigned_bell(self):
+        """ Called whenever Wheatley is assigned to a bell. """
+        self._make_active()
+
+    def on_call(self, call_name):
+        if call_name == "Look to" and USER_NAME in self._tower.assignments.values():
```

Comment:
With the change to user ids this should now be
```suggestion
        if call_name == "Look to" and USER_ID in self._tower.assignments.values():
```

Reply:
Nice spot - fixed!